### PR TITLE
add pre-auth js code

### PR DESCRIPTION
**Pull request by brianshen1990**
add pre-auth js code

Please also review the doc side PR https://github.com/Shopify/shopify-dev/pull/46529

### CODE REVIEW

File: javascript/example-customer-account--pre-auth--ts/extensions/customer-account-pre-auth-loyalty/shopify.extension.toml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# [START config.setup-targets]
+[[extensions.targeting]]
+module = "./src/BlockLoyaltyExtension.ts"
+target = "customer-account.order-status.block.render"
+# [END config.setup-targets]
+
+# [START config.api-access]
+[extensions.capabilities]
+# api_access = true
+# [END config.api-access]
```
Comment (jrrafols) [inline javascript/example-customer-account--pre-auth--ts/extensions/customer-account-pre-auth-loyalty/shopify.extension.toml:17]: It doesn't look like you actually use this tag or setting anywhere in the tutorial. I think this should be safe to delete

File: javascript/example-customer-account--pre-auth--ts/extensions/customer-account-pre-auth-loyalty/src/BlockLoyaltyExtension.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import {
+  Button,
+  Card,
+  InlineStack,
+  extension,
+} from "@shopify/ui-extensions/customer-account";
+
+export default extension(
+  "customer-account.order-status.block.render",
+  // [START order-status-block.setup-target]
```
Comment (jrrafols) [inline javascript/example-customer-account--pre-auth--ts/extensions/customer-account-pre-auth-loyalty/src/BlockLoyaltyExtension.ts:11]: Should this be `END`?

File: javascript/example-customer-account--pre-auth--ts/extensions/customer-account-pre-auth-note/shopify.extension.toml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+[[extensions.targeting]]
+module = "./src/MenuActionModalExtension.ts"
+target = "customer-account.order.action.render"
+# [END config.setup-targets]
+
+# [START config.api-access]
+[extensions.capabilities]
+# api_access = true
+# [END config.api-access]
```
Comment (jrrafols) [inline javascript/example-customer-account--pre-auth--ts/extensions/customer-account-pre-auth-note/shopify.extension.toml:21]: Similar comment here, I don't think this is needed?

File: javascript/example-customer-account--pre-auth--ts/extensions/customer-account-pre-auth-note/src/MenuActionModalExtension.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        root.createComponent(Form, { onSubmit: () => {
+          try {
+              // [START menu-action-modal.make-request]
+              // make a request to the server to add a note
+              // [END menu-action-modal.make-request]
+          } catch(error) {
+              // toast api if possible
+              console.log(error);
+          } finally {
+              // api.storage.write("orderNote", note);
```
Comment (jrrafols) [inline javascript/example-customer-account--pre-auth--ts/extensions/customer-account-pre-auth-note/src/MenuActionModalExtension.ts:32]: Just wanted to double-check that the comments were left intentionally?

**Review by jrrafols — APPROVED** [review]
Left a couple notes about things that could be cleaned up, but otherwise LGTM. Approving assuming comments will be addressed.

File: javascript/example-customer-account--pre-auth--ts/extensions/customer-account-pre-auth-loyalty/shopify.extension.toml (hunk 1 above)
Comment (brianshen1990) [reply javascript/example-customer-account--pre-auth--ts/extensions/customer-account-pre-auth-loyalty/shopify.extension.toml]: updated~

File: javascript/example-customer-account--pre-auth--ts/extensions/customer-account-pre-auth-note/shopify.extension.toml (hunk 3 above)
Comment (brianshen1990) [reply javascript/example-customer-account--pre-auth--ts/extensions/customer-account-pre-auth-note/shopify.extension.toml]: updated~

**Pull request merged**